src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/RemoteJenkinsServer.java

```diff
@@ -27,116 +30,152 @@
+    }
 
-        this.address = new URL(address);
-        this.displayName = displayName.trim();
-        this.hasBuildTokenRootSupport = hasBuildTokenRootSupport;
+    @DataBoundSetter
+    public void setDisplayName(String displayName)
+    {
+        this.displayName = trimToEmpty(displayName);
+    }
 
-        // Holding on to both of these variables for legacy purposes. The seemingly 'dirty' getters for these properties
-        // are for the same reason.
-        this.username = "";
-        this.apiToken = "";
+    @DataBoundSetter
+    public void setHasBuildTokenRootSupport(boolean hasBuildTokenRootSupport)
+    {
+        this.hasBuildTokenRootSupport = hasBuildTokenRootSupport;
+    }
 
-        // this.auth = new Auth(auth);
-        this.auth.replaceBy(new Auth(auth));
+    @DataBoundSetter
+    public void setAuth2(Auth2 auth2)
+    {
+        this.auth2 = (auth2 != null) ? auth2 : new NoneAuth();
+    }
 
+    @DataBoundSetter
+    public void setAddress(String address)
+    {
+        this.address = address;
     }
 
     // Getters
 
-    public Auth[] getAuth() {
-        return auth.toArray(new Auth[this.auth.size()]);
-    }
-
+    @CheckForNull
     public String getDisplayName() {
         String displayName = null;
 
         if (this.displayName == null || this.displayName.trim().equals("")) {
-            displayName = this.getAddress().toString();
+            if (address != null) displayName = address;
+            else displayName = null;
         } else {
             displayName = this.displayName;
         }
         return displayName;
     }
 
-    public URL getAddress() {
-        return address;
+    public boolean getHasBuildTokenRootSupport() {
+        return hasBuildTokenRootSupport;
     }
 
-    public boolean getHasBuildTokenRootSupport() {
-        return this.hasBuildTokenRootSupport;
+    @CheckForNull
+    public Auth2 getAuth2() {
+        migrateAuthToAuth2();
+        return auth2;
+    }
+
+    /**
+     * Migrates old <code>Auth</code> to <code>Auth2</code> if necessary. 
+     * @deprecated since 2.3.0-SNAPSHOT - get rid once all users migrated
+     */
+    private void migrateAuthToAuth2() {
+        if(auth2 == null) {
+            if(auth == null || auth.size() <= 0) {
+                auth2 = new NoneAuth(); 
+            } else {
+                auth2 = Auth.authToAuth2(auth);
+            }
+        }
+        auth = null;
+    }
+
+    @CheckForNull
+    public String getAddress() {
+        return address;
     }
 
     @Override
     public DescriptorImpl getDescriptor() {
         return (DescriptorImpl) super.getDescriptor();
     }
 
+    /**
+     * @return the remote server address
+     * @throws RuntimeException
+     *             if the address of the remote server was not set
+     */
+    @Nonnull
+    public String getRemoteAddress() {
+        if (address == null) {
+            throw new RuntimeException("The remote address can not be empty.");
```

Comment:
In your case it will be empty if the class is restored from the disk after the migration

Reply:
https://github.com/sap-production/parameterized-remote-trigger-plugin/pull/22